gcp: allow non-bootstrap kubeconfig

awly opened the pull request:
**What this PR does / why we need it**:
Needed for https://github.com/kubernetes/community/pull/2022
This change lets us generate a non-bootstrap kubeconfig with exec plugin for authn.
The plugin does TLS bootstrapping internally.

**Special notes for your reviewer**:
Defaults when no new env vars are set will behave same as before this change.
`KUBELET_AUTH_TYPE` should never be `tls-auth` in practice, but leaving it there just in case.

**Release note**:
```release-note
NONE
```

awly commented:
/assign @vishh @mikedanese

awly commented:
/retest

mikedanese commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -893,7 +893,8 @@ function create-kubelet-kubeconfig() {
   fi
   echo "Creating kubelet kubeconfig file"
-  cat <<EOF >/var/lib/kubelet/bootstrap-kubeconfig
+  if [[ "${CREATE_BOOTSTRAP_KUBECONFIG:-true}" == "true" ]]; then
```
I'd suggest sourcing this from the metadata server in a new attribute. It would give us a bit more flexibility later. What do you think?

mikedanese commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -913,6 +914,36 @@ contexts:
+    user: ${KUBELET_AUTH_TYPE:-tls-auth}
+  name: default-context
+current-context: default-context
+kind: Config
```
put kind up by apiversion

mikedanese commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -913,6 +914,36 @@ contexts:
+  name: default-context
+current-context: default-context
+kind: Config
+preferences: {}
```
omit this.

Commit:
gcp: allow non-bootstrap kubeconfig

The regular kubeconfig is fetched from metadata when
CREATE_BOOTSTRAP_KUBECONFIG==false.

We will experiment with an exec plugin that does TLS bootstrapping
internally: #61803

awly commented:
PTAL @mikedanese

awly commented:
/retest

Commit:
Allow fetching bootstrap-kubeconfig from VM metadata

awly commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -913,6 +914,10 @@ contexts:
 EOF
+  else
+    echo "Fetching kubelet kubeconfig file from metadata"
+    get-metadata-value "instance/attributes/kubeconfig" >/var/lib/kubelet/kubeconfig
```
Good idea, I think we will. Added as 3rd option.

mikedanese commented:
/lgtm
